Add Byte, Int16, Int32, Int64, SByte, UInt16, UInt32 and UInt64 tests

hughbe opened the pull request:
- Take advantage of xunit
- Refactor and add tests for Parse and ToString

Contributes to #6286
Cherry picked from #5490
Helps test dotnet/coreclr#3995

ianhays commented on src/System.Runtime/tests/System/Int16.cs:
```diff
@@ -38,255 +36,237 @@ public static void TestMinValue()
+        yield return new object[] { "123", NumberStyles.HexNumber, nullFormat, (short)0x123 };
+        yield return new object[] { "abc", NumberStyles.HexNumber, nullFormat, (short)0xabc };
+        yield return new object[] { "1000", NumberStyles.AllowThousands, nullFormat, (short)1000 };
+        yield return new object[] { "(123)", NumberStyles.AllowParentheses, nullFormat, (short)-123 }; // Parentheses = negative
```
Why not just use null instead of ```nullFormat```?

hughbe replied:
I wanted to make it clear that this was the IFormatProvider parameter, but I can undo this if you want

ianhays commented on src/System.Runtime/tests/System/Int64.cs:
```diff
@@ -38,261 +36,230 @@ public static void TestMinValue()
+    [InlineData((long)234, (long)456, -1)]
+    [InlineData((long)234, long.MaxValue, -1)]
+    [InlineData((long)234, null, 1)]
+    public static void TestCompareTo(long i, object value, long expected)
```
Might be good to keep at least one negative value.

hughbe replied:
We did right?
`[InlineData((long)234, (long)-123, 1)]`

Or do you mean comparing a negative long to a positive one or a more negative one?

ianhays replied:
Oops, I missed that :) That's fine.

ianhays commented:
A few minor nits but otherwise this LGTM.

hughbe commented:
@ianhays  thanks for the review. I updated the PR acccording to your feedback, and added some more tests for Parse (I'm working on a PR over in coreclr improving the perf of parsing numbers so these helped)

I had to rebase to fix a merge conflict, sorryy

hughbe commented:
Test Innerloop OSX Debug Build and Test
Test Innerloop OSX Release Build and Test
